File: pset1/mario_folder/less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // User input
    printf("Enter desired half-pyramid height.\n");
    int height = get_int();

    // Data validation
    while ( height < 0 || height > 23 )
    {
        printf("Retry: ");
        height = get_int();
    }

    // Outer loop determines number of output lines
    for ( int i = 0; i < height; i++ )
    {
        // First inner loop prints decreasing number of spaces
        for ( int j = height; j > i + 1; j-- )
        {
            printf(" ");
        }

        // Second inner loop prints out increasing number of octothorpes
        for ( int k = 0; k < i + 2; k++ )
        {
            printf("#");
        }

        // New line after inner loops complete
        printf("\n");
    }
}

File: pset6/mario/mario.py
from cs50 import get_int

# user input
print("Height: ", end='')
max_height = get_int()

# data validation
while (max_height < 0 or max_height > 23):
    print("Height: ", end='')
    max_height = get_int()

# outer loop determines number of levels
for i in range(0, max_height):

    # inner loop #1 print out spaces
    for j in range(max_height - 1 - i, 0, -1):
        print(" ", end='')

    # inner loop #2 print out hashtags
    for k in range(0, i + 1):
        print("#", end='')

    # print out two spaces
    print("  ", end='')

    # inner loop #3 print out hashtags
    for i in range(0, i + 1):
        print("#", end='')

    # new line
    print('')